### PR TITLE
Fix the entitlements team cache when there are multiple instances of GitHub in use.

### PR DESCRIPTION
**Pull request by chrisgavin**
I think currently the team cache gets confused when you have multiple GitHub Enterprise Server instances (or one Enterprise instance and DotCom) that have an organization and team with the same name.

To avoid this, I believe we need to include the base address of the instance as part of the cache structure. This ensures each instance has its own cache.

### CODE REVIEW

**Comment by chrisgavin** [comment]
The acceptance tests are also failing on main in the same way, so I don't think they're related to the change.

File: lib/entitlements/backend/github_team/service.rb [hunk 1]
```diff
@@ -35,8 +35,9 @@ class TeamNotFound < RuntimeError; end
         def initialize(addr: nil, org:, token:, ou:, ignore_not_found: false)
           super
           Entitlements.cache[:github_team_members] ||= {}
-          Entitlements.cache[:github_team_members][org] ||= {}
-          @team_cache = Entitlements.cache[:github_team_members][org]
+          Entitlements.cache[:github_team_members][addr] ||= {}
+          Entitlements.cache[:github_team_members][addr][org] ||= {}
+          @team_cache = Entitlements.cache[:github_team_members][addr][org]
```
Comment (jasonmacgowan) [inline lib/entitlements/backend/github_team/service.rb:40]: 🎉 Thank you for this!

There is a method on the base GitHub service class use use to handle this collision for org membership, let's use that instead for consistency.

```suggestion
          Entitlements.cache[:github_team_members][org_signature] ||= {}
          @team_cache = Entitlements.cache[:github_team_members][org_signature]
```

**Review by jasonmacgowan — CHANGES_REQUESTED** [review]

**Comment by jasonmacgowan** [comment]
https://github.com/github/entitlements-github-plugin/pull/32 should fix the acceptance test errors

**Comment by jasonmacgowan** [comment]
@chrisgavin can you also bump the version so we can publish the gem when it merges :bow:

**Review by jasonmacgowan — APPROVED** [review]

**Pull request merged**